pkg/virtctl/imageupload/imageupload.go

```diff
@@ -641,7 +670,11 @@ func setDefaultInstancetypeLabels(target metav1.Object) {
 	}
 }
 
-func createUploadDataVolume(client kubecli.KubevirtClient, namespace, name, size, storageClass, accessMode, volumeMode string, archiveUpload bool) (*cdiv1.DataVolume, error) {
+func createUploadDataVolume(
+	client kubecli.KubevirtClient,
+	namespace, name, size, storageClass, accessMode, volumeMode string,
+	defaultInstancetype, defaultInstancetypeKind, defaultPreference, defaultPreferenceKind string,
+	archiveUpload, forceBind bool) (*cdiv1.DataVolume, error) {
```

Comment:
The PR is getting rid if the globals, which is good, but adds very many parameters, which decreases the readability again. I am not sure, but I would expect that if sonar cloud would be still enabled, it might even block your PR for that reason. Searching for https://www.google.com/search?q=code+smell+too+many+parameters might give you also an idea.

Reply:
I'll refactor it to have less params.